src/Microsoft.AspNetCore.Mvc.Abstractions/ModelBinding/EnumGroupAndName.cs

```diff
@@ -28,7 +38,39 @@ public EnumGroupAndName(string group, string name)
             }
 
             Group = group;
-            Name = name;
+
+            _stringLocalizer = null;
+            _fieldInfo = null;
+            _displayAttribute = null;
+            _name = name;
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the EnumGroupAndName structure.
+        /// </summary>
+        /// <param name="group">The group name.</param>
+        /// <param name="stringLocalizer">The <see cref="IStringLocalizer"/> to localize with.</param>
+        /// <param name="fieldInfo">The <see cref="FieldInfo"/> to use for display.</param>
+        public EnumGroupAndName(
+            string group,
+            IStringLocalizer stringLocalizer,
+            FieldInfo fieldInfo)
+        {
+            if (group == null)
+            {
+                throw new ArgumentNullException(nameof(group));
+            }
+
+            if (fieldInfo == null)
+            {
+                throw new ArgumentNullException(nameof(fieldInfo));
+            }
+
+            Group = group;
+            _stringLocalizer = stringLocalizer;
+            _fieldInfo = fieldInfo;
+            _name = null;
```

Comment:
Only use `_fieldInfo.Name` now. Can store that in `_name` and remove the `_fieldInfo` field.